### PR TITLE
ST-2/Patch- duplicate harvest use and permission

### PR DESCRIPTION
**Pull request by Tbrid**

### CODE REVIEW

File: packages/api/src/controllers/logController.js [hunk 1]
```diff
@@ -115,7 +115,10 @@ const logController = {
       const { name } = req.body;
       const trx = await transaction.start(Model.knex());
       try {
-        const check = await HarvestUseTypeModel.query().where({ farm_id, harvest_use_type_name: name }).first();
+        const check = await HarvestUseTypeModel.query()
+          .where(builder => builder.where('farm_id', farm_id).orWhere('farm_id', null))
```
Comment (beetai) [inline packages/api/src/controllers/logController.js:119]: what is the builder for? how is this different from simply calling `.where('farm_id', farm_id)`?

File: packages/webapp/src/containers/Task/TaskComplete/HarvestComplete/AddHarvestUseType.js [hunk 2]
```diff
@@ -25,17 +36,18 @@ export default function AddHarvestUseTypeModal({ dismissModal }) {
...
-        style={{ marginBottom: '12px', marginTop: '8px' }}
+        style={{ marginTop: '8px' }}
         label={t('TASK.DESCRIBE_HARVEST_USE')}
-        onChange={(e) => setHarvestUseName(e.target.value)}
+        onChange={(e) => onHarvestUseInputChange(e)}
       />
+      {duplicateError && <Error>{t('TASK.HARVEST_USE_ALREADY_EXISTS')}</Error>}
     </ModalComponent>
   );
 }
```
Comment (beetai) [inline packages/webapp/src/containers/Task/TaskComplete/HarvestComplete/AddHarvestUseType.js:53]: eof whitespace

File: packages/api/src/controllers/logController.js (hunk 1 above)
Comment (Tbrid) [reply packages/api/src/controllers/logController.js:119]: want to get the default harvest uses too

Comment (Tbrid) [reply packages/api/src/controllers/logController.js:119]: farm_id is always not null in the request

Comment (beetai) [reply packages/api/src/controllers/logController.js:119]: isn't that covered by `.orWhere('farm_id', null)`? or am i mistaken

Comment (beetai) [reply packages/api/src/controllers/logController.js:119]: just wondering, aren't default harvest uses covered by `.ofWhere('farm_id', null)`?

Comment (Tbrid) [reply packages/api/src/controllers/logController.js:119]: what's ofWhere?

Comment (beetai) [reply packages/api/src/controllers/logController.js:119]: oeWhere***

Comment (Tbrid) [reply packages/api/src/controllers/logController.js:119]: Yes

**Review by beetai — APPROVED** [review]

**Pull request merged**